CI-API.Data/Repository/AdminRepository.cs

```diff
@@ -330,5 +236,529 @@ private JsonResult updateAddUser(User userHasTobeUpdated,UserDetailViewModel use
+                    {
+                        missions = AllMission,
+                        missionThemes = AllMissionThemes,
+                        missionSkills = AllMissionSkills,
+
+                    };
+
+                    return new JsonResult(new apiResponse<LandingPageViewModel> { StatusCode = responseStatusCode.Success, Data = landingPageViewModel, Result = true });
+                }
+            }
+            catch
+            {
+                return new JsonResult(new apiResponse<string> { Message = ResponseMessages.InternalServerError, StatusCode = responseStatusCode.BadRequest, Result = false });
+
+            }
+        }
+        #endregion
+
+        #region GetMissionDataFromId
+        public async Task<JsonResult> getMissionDataFromId(long? missionId)
+        {
+            try
+            {
+                Mission? missionData = await Task.FromResult(cIDbContext.Missions.Where(M => M.MissionId == missionId).FirstOrDefault());
+                List<MissionDocument> documentsForMission = await Task.FromResult(cIDbContext.MissionDocuments.Where(MD => MD.MissionId == missionId).ToList());
+                List<MissionMedium> mediaForMission = await Task.FromResult(cIDbContext.MissionMedia.Where(MM => MM.MissionId == missionId).ToList());
+                List<MissionSkill> skillForMission = await Task.FromResult(cIDbContext.MissionSkills.Where(MS => MS.MissionId == missionId).ToList());
+                if (missionData?.MissionType == StaticCode.GoalMission)
+                {
+                    GoalMission? goalValueForMission = await Task.FromResult(cIDbContext.GoalMissions.Where(GM => GM.MissionId == missionId).FirstOrDefault());
+                    LandingPageViewModel missionDataForEdit = new()
+                    {
+                        mission = missionData,
+                        missionDocuments = documentsForMission,
+                        missionMedia = mediaForMission,
+                        goalMissions = goalValueForMission,
+                        missionSkills = skillForMission
+                    };
+                    return new JsonResult(new apiResponse<LandingPageViewModel> { Message = ResponseMessages.MissionAddedSuccessfully, StatusCode = responseStatusCode.Success, Data = missionDataForEdit, Result = true });
+                }
+                else
+                {
+                    LandingPageViewModel missionDataForEdit = new()
+                    {
+                        mission = missionData,
+                        missionDocuments = documentsForMission,
+                        missionMedia = mediaForMission,
+                        missionSkills = skillForMission
+                    };
+                    return new JsonResult(new apiResponse<LandingPageViewModel> { Message = ResponseMessages.MissionAddedSuccessfully, StatusCode = responseStatusCode.Success, Data = missionDataForEdit, Result = true });
```

Comment:
this result just returning mission data, don't use message - ResponseMessages.MissionAddedSuccessfully

Reply:
changes done